LibOS/shim/src/sys/shim_socket.c

```diff
@@ -720,7 +724,7 @@ int shim_do_connect (int sockfd, struct sockaddr * addr, int addrlen)
         char * spath = saddr->sun_path;
         struct shim_dentry * dent;
 
-        if ((ret = path_lookupat(NULL, spath, LOOKUP_CREATE, &dent)) < 0)
+        if ((ret = path_lookupat(NULL, spath, LOOKUP_CREATE, &dent, &socket_builtin_fs)) < 0)
```

Comment:
I don't think this is right. You can't possibly from this dentry from socket_builtin_fs because socket_builtin_fs is never mounted. Just look up the dentry like a regular path.

Reply:
Good catch.  For unix sockets they should be reachable on the main FS.  I'll fix this.